opta/core/cloud_provider.py

```diff
@@ -0,0 +1,52 @@
+import boto3
+from botocore.config import Config
+from botocore.exceptions import ClientError
+
+from opta.core.terraform2 import TerraformFile
+from opta.layer2 import Layer
+
+
+class CloudProvider:
+    def state_storage(self, layer: Layer) -> str:
+        return f"opta-tf-state-{layer.org_name}-{layer.name}"
+
+    def verify_storage(self, layer: Layer) -> bool:
+        raise NotImplementedError()
+
+    def add_provider_config(self, file: TerraformFile, layer: Layer) -> None:
+        raise NotImplementedError()
+
+
+class AWSProvider(CloudProvider):
```

Comment:
what do you think of breaking up the cloud implementation in it's own class from the beginning?
mostly to avoid having large files and a good separation between cloud specific impl / cloud agnostic

Reply:
Once I dive deeper into the cloud specific code (such as remote state storage), I will refactor the python modules/classes and organize them better.